crates/typst/src/foundations/calc.rs

```diff
@@ -866,45 +972,77 @@ pub fn div_euclid(
 /// value of the divisor if the dividend is much smaller in magnitude than the divisor
 /// and the dividend is negative. This only applies for floating point inputs.
 ///
+/// In addition, this can error if given a [`{decimal}`]($decimal) input and
+/// the dividend is too small in magnitude compared to the divisor.
+///
```

Comment:
I do wonder if we should just return some specific sentinel value (e.g. the divisor itself) when this happens instead of raising an error (given the warning above for floats). Not sure what would be mathematically correct though, and I guess ensuring the range of the returned remainder is 0.0 <= r < divisor is more important.

Same applies to `calc.rem`.

Reply:
Not sure. We can always remove the error later.